src/providers/articles.ts

```diff
@@ -0,0 +1,25 @@
+import { default as express } from 'express';
+import { articleManager } from '../services/article-manager.js';
+import { Article } from '../types/article.js';
+
+// Returns an array of Articles as JSON
+export async function getArticlesAsJSON(
+  request: express.Request,
+  response: express.Response,
+  next: express.NextFunction
+) {
+  if (request.headers.accept && request.header('Accept') === 'application/json') {
+    // FIXME: The object returned here should be reduced first.
+    const articles: Array<Article> = [];
+    for (const article of articleManager.values()) {
+      articles.push(article);
+    }
```

Comment:
You can also use `Array.from(articleManager.values())` or spread `[...articleManager.values()]` to convert an iterator to an array